### PR TITLE
remove some unused functions as reported by cppcheck

### PR DESCRIPTION
**Pull request by rongou**
After removing `gpu_hist`, I ran cppcheck to see if it can find more dead code, but it found a few random functions.

@RAMitchell @trivialfis @hcho3

### CODE REVIEW

File: src/common/bitmap.h [hunk 1]
```diff
@@ -47,7 +47,7 @@ struct BitMap {
     for (bst_omp_uint i = 0; i < nsize; ++i) {
       uint32_t res = 0;
       for (int k = 0; k < 32; ++k) {
-        int bit = vec[(i << 5) | k];
+        uint32_t bit = vec[(i << 5) | k];
```
Comment (rongou) [inline src/common/bitmap.h]: cppcheck was complaining about `shifting signed 32-bit value by 31 bits is undefined behaviour`.

File: src/learner.cc [hunk 2]
```diff
@@ -31,13 +31,13 @@ const char* kMaxDeltaStepDefaultValue = "0.7";
 
 inline bool IsFloat(const std::string& str) {
   std::stringstream ss(str);
-  float f;
```
Comment (rongou) [inline src/learner.cc]: Uninitialized variable

File: src/learner.cc [hunk 3]
```diff
@@ -31,13 +31,13 @@ const char* kMaxDeltaStepDefaultValue = "0.7";
...
+  float f{};
   return !((ss >> std::noskipws >> f).rdstate() ^ std::ios_base::eofbit);
 }
 
 inline bool IsInt(const std::string& str) {
   std::stringstream ss(str);
-  int i;
```
Comment (rongou) [inline src/learner.cc]: Uninitialized variable

File: src/common/bitmap.h (hunk 1 above)
Comment (trivialfis) [reply src/common/bitmap.h]: Good to know. Thanks.

**Pull request merged**